### ISSUE TITLE
only aruco take off error

### ISSUE THREAD

**Issue by youngjae41** [CopterExpress/clover#492]
Hello Long time no see @okalachev 

I am using px4 1.12.3 version firmware. Optical flow is set to false and I am testing using ipython

<arg name="force_init" default="true"/> 
<arg name="simulator" default="true"/> 

set it like this

navigate(x=0, y=0, z=1.0, speed=0.5, frame_id='body', auto_arm=True)

Give the take off command. The gas initially rises and then loses position.

aruco.launch file

    <arg name="aruco_detect" default="true"/>
    <arg name="aruco_map" default="false"/>
    <arg name="aruco_vpe" default="false"/>
    <arg name="placement" default="floor"/> 
    <arg name="length" default="0.22"/> 
    <arg name="map" default="map.txt"/> 
    
    Doesn't it work if I don't use aruco_map?
    
    And another bug is that if you don't set aruco_map to True, rviz doesn't output image subjects to aruco_dectect.

**Comment by okalachev** [comment]
Hi! You can't fly in position mode with optical flow turned off and aruco vpe turned off, since the drone doesn't have the position source.

**Comment by youngjae41** [comment]
@okalachev thanks ~! I'm using ipython3 control but land position is strange 
The command is using Land() and using aruco.

**Comment by okalachev** [comment]
What do you mean land position is strange? Please, provide more details on that and, if possible, the flight logs.

**Comment by youngjae41** [comment]
@okalachev It seems to be a camera problem. Existing clover rpi camera outputs 40 fps, but jetson camera outputs 30 fps max. Could this be why the optical flow isn't working properly?

**Comment by youngjae41** [comment]
@okalachev aruco.launch aruco_detect        <param name="use_map_markers" value="true"/> If not set to false, aruco detection will not work.

**Comment by okalachev** [comment]
@youngjae41 

>It seems to be a camera problem. Existing clover rpi camera outputs 40 fps, but jetson camera outputs 30 fps max. Could this be why the optical flow isn't working properly?

No, there is no hard requirement to optical flow camera fps to work, 30 FPS is OK.

>aruco.launch aruco_detect If not set to false, aruco detection will not work.

You can use an automated tool that checks your Clover's configuration and prints a full report: `rosrun clover selfcheck.py`.

**Comment by youngjae41** [comment]
@okalachev Sorry, misdirection. in the latest version

  <param name="use_map_markers" value="true"/> <- set to false, aruco detection will not work.

**Comment by okalachev** [comment]
You can use the above checking tool to diagnose some problems. Also there is log in `/var/log/clover.log` file, if aruco detection falls, there would be errors.

**Comment by lgh5054** [comment]
@okalachev Hello, is there a way to land on the Aruco Maker I want?

**Comment by okalachev** [comment]
You can use `aruco_X` frame for that, like, `navigate(x=0,y=0,z=0,frame_id='aruco_35')`.

**Comment by lgh5054** [comment]
@okalachev Entering Land() RTLs to the place where the aircraft first takek off. Has this ever happened?

**Comment by okalachev** [comment]
@lgh5054, can you please reformulate your question? I don't get it.

**Comment by lgh5054** [comment]
@okalachev When entering land(), it does not land at the current location, but RTLs to the home point from which it first took off.

**Comment by okalachev** [comment]
No, it should not work this way. Maybe it just drifted away for some reason when it was landing?

**Comment by lgh5054** [comment]
@okalachev i'm using v 1.14.0 relrease firmware

**Comment by okalachev** [comment]
Sorry, this code was not written by us, you can try to contact its author https://t.me/tenessinum to get it.

**Comment by lgh5054** [comment]
@okalachev Hello Okalachev Found the problem.

When operating optical flow, even if you hold the drone with your hand and move its position, rviz will initialize it to 0,0,0 coordinates.

optical flow problem or px4 1.14.0 firmware problem?

**Comment by okalachev** [comment]
Closed as stale.

**Comment by okalachev** [comment]
Closed as stale.
